File: BackEnd/ltddnc_backend/ltddnc_backend/Model/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ltddnc_backend.Model
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int Stock { get; set; } = 0;
        public double UnitPrice { get; set; }
        public int State { get; set; } = 1;
        public int? IdCategory { get; set; }
        public Category Category { get; set; }
        public string? Image { get; set; }
    }
}

File: lib/models/product.dart
import 'dart:convert';

Product productFromJson(String str) => Product.fromJson(json.decode(str));
String productToJson(Product data) => json.encode(data.toJson());

class Product {
  final int? id;
  final String? name;
  final double? unitPrice;
  final String? image;
  final String? description;
  final int? stock;
  final int? state;
  final int? idCategory;
  final double? avgRating;
  final String? createdDate;

  Product(
      {this.id,
      this.name,
      this.unitPrice,
      this.image,
      this.description,
      this.stock,
      this.state,
      this.idCategory,
      this.avgRating,
        this.createdDate});

  Map<String, dynamic> toJson() => {
        "id": id,
        "name": name,
        "unitPrice": unitPrice,
        "image": image,
        "description": description,
        "stock": stock,
        "state": state,
        "idCategory": idCategory,
        "avgRating": avgRating,
        "createdDate": createdDate
      };

  factory Product.fromJson(Map<String, dynamic> json) => Product(
      id: json["id"],
      name: json["name"],
      unitPrice: json["unitPrice"],
      image: json["image"],
      description: json["description"],
      stock: json["stock"],
      state: json["state"],
      idCategory: json["idCategory"],
      avgRating: json["avgRating"],
      createdDate: json["createdDate"]);
}
